Readability.js

```diff
@@ -721,11 +721,11 @@ Readability.prototype = {
           }
         }
 
-        // Remove empty DIV, SECTION, and HEADER nodes
+        // Remove DIV, SECTION, and HEADER nodes without any content(e.g. text, image, video, or iframe).
```

Comment:
The root cause of this bug is choosing wrong top candidate(currently, the algorithm select `<div class="section-content">` that belongs to `<div class="section-3">` in the below dom tree as the top candidate).

```html
<div class="postArticle-content">
  <div class="section-1">
    <div class="section-divider"><hr class="section-divider" /></div>
    <div class="section-content"></div>
  </div>
  <div class="section-2">
    <div class="section-divider"><hr class="section-divider" /></div>
    <div class="section-content"></div>
  </div>
  <div class="section-3">
    <div class="section-divider"><hr class="section-divider" /></div>
    <div class="section-content"></div>
  </div>
  <div class="section-4">
    <div class="section-divider"><hr class="section-divider" /></div>
    <div class="section-content"></div>
  </div>
  <div class="section-5">
    <div class="section-divider"><hr class="section-divider" /></div>
    <div class="section-content"></div>
  </div>
</div>
```

And the patch is trying to remove the `<div class="section-divider">` node that belongs to `<div class="section-3">`, and it helps the [logic added for nytimes pages][nytimes-logic] select correct top candidate(that is `<div class="postArticle-content">`).

I would say removing containers that ONLY contain `<br>` or `<hr>` might not be a wrong idea. Because it will only remove the (useless?) nodes like below
```html
<div><hr /></div>
```
```html
<div><br /></div>
```
and it won't remove the (meaningful) nodes like below
```html
<div>
  <h1>Lorem</h1>
  <hr />
  <p>Tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum</p>
  <br />
</div>
```

Removing the useless `<div class="section-divider">` nodes in the above dom tree is a simple approach to fix the issue and it also seems to make sense. If you also think we could take the solution, then we could change and update current implementation in the patch.

What do you think? Or @gijsk you have other good ideas to fix the issue. (I'm think of it)

[nytimes-logic]: https://github.com/mozilla/readability/blob/master/Readability.js#L925-L931